itou/users/factories.py

```diff
@@ -134,6 +134,12 @@ def nir(self):
         validate_nir(nir)
         return nir
 
+    @factory.post_generation
+    def without_profile(obj, create, extracted, **kwargs):
+        if extracted and create:
+            obj.jobseeker_profile.delete()
+            obj.jobseeker_profile = None
+
```

Comment:
J'ai le cerveau cramé mais en fait je pense que ce que tu voulais faire (enfin ce que j'ai compris c'était permettre `jobseeker_profile__field=value`) dans le `JobSeekerProfileFactory._create` pourrais fonctionner ici et on ne devrais pas tomber dans le cas où on MAJ de la donnée existante avant l'appel à la _factory_.
```suggestion
    @factory.post_generation
    def jobseeker_profile(obj, create, extracted, **kwargs):
        for k, v in kwargs.items():
            setattr(obj.jobseeker_profile, k, v)

        if extracted is False:
            obj.jobseeker_profile.delete()
            obj.jobseeker_profile = None
        elif create:
            obj.jobseeker_profile.save(update_fields=kwargs.keys())
```

Reply:
Je ne suis pas sur qu'on cherchait à faire le `jobseeker_profile__field=value` spécifiquement et il me semblait qu'il fonctionnait déjà, alors cerveau contre-cramé sûrement mais je ne vois pas ce que cette suggestion approte de plus, je dois être dans la continuité de cette petite semaine.

Reply:
Effectivement c'était mon interprétation et pas ça qui était voulu :).

L'avantage de pouvoir passer le contexte c'est que si on a besoin de modifier 1 (ou plusieurs) champs sur le  `User()` et le `JobSeekerProfile()` on peux le faire en 1 étape au lieu de 2, ça devrais aussi éviter de devoir utiliser la mécanique du `without_profile=True` dés qu'on tombe dans ce cas précis.
Après ça ne gère pas les champs enfants comme le ferais une vraie _factory_ mais on en a surement pas besoin, et si on en arrivais là alors a priori faudrait passer par les signaux et le lien que mentionnais Xavier.